prototype/core/impl/src/main/java/org/eclipse/sensinact/prototype/model/nexus/impl/emf/EMFUtil.java

```diff
@@ -175,4 +178,17 @@ public static int getContainerVersion(EStructuralFeature feature) {
         return getVersion((EClass) feature.eContainer());
     }
 
+    public static Object convertToTargetType(EClassifier targetType, Object o) {
+
+        if (o == null) {
+            return targetType.getInstanceClass().isPrimitive() ? targetType.getDefaultValue() : o;
+        } else {
+            EClassifier type = typeMap.get(o.getClass());
+            if (type == null) {
+                throw new IllegalArgumentException("Unknown data type " + o.getClass());
+            }
+            String string = type.getEPackage().getEFactoryInstance().convertToString((EDataType) type, o);
```

Comment:
This seems like an unneceesary step or do we want to use this to create a deep copy of the underlying object?

Reply:
Are you suggesting that we should do a type check and short circuit at the start?

Reply:
The main reason for all of this was to support setting admin/location as a String and parsing it appropriately, but also to allow for simple Long/Integer processing

Reply:
This seems like an unneceesary step or do we want to use this to create a deep copy of the underlying object?

Reply:
Sorry, my mistake. I missed the distinction between type and targetType. In this light it makes sense.